modules/security/keyvault/main.bicep

```diff
@@ -0,0 +1,84 @@
+@description('Deployment Location')
+param location string
+
+@description('Name of the Key Vault')
+param name string = 'keyvault-${uniqueString(resourceGroup().id)}'
+
+@description('Subnet ID for the Key Vault')
+param subnetID string = ''
+
+@description('The tenant ID where the Key Vault is deployed')
+param tenantId string = subscription().tenantId
+
+@description('Enable VNet Service Endpoints for Key Vault')
+param enableVNet bool = false
+
+@description('List of RBAC policies to assign to the Key Vault')
+param rbacPolicies array = []
+
+@allowed([ 'new', 'existing'])
+@description('Specifies whether to create a new Key Vault or use an existing one. Use "new" to create a new Key Vault or "existing" to use an existing one.')
+param newOrExisting string = 'new'
+
+@description('Enable role assignment for the Key Vault')
+param assignRole bool = true
```

Comment:
Why not offer customization of what role is used? This locks it down where I'd have to run it again to assign Key access.

Reply:
had to break things out into a second module for the double for loop, but was able to get this done.